processor/metricstransformprocessor/datapoint_aggregation.go

```diff
@@ -0,0 +1,248 @@
+func (mtp *metricsTransformProcessor) computeInt64(points []*metricspb.Point, aggrType AggregationType) *metricspb.Point_Int64Value {
+	intVal := int64(0)
+	if len(points) > 0 {
+		intVal = points[0].GetInt64Value()
+	}
+	for _, p := range points[1:] {
+		if aggrType == Sum || aggrType == Mean {
+			intVal += p.GetInt64Value()
+		} else if aggrType == Max {
+			intVal = mtp.maxInt64(intVal, p.GetInt64Value())
+		} else if aggrType == Min {
+			intVal = mtp.minInt64(intVal, p.GetInt64Value())
+		}
+	}
+	if aggrType == Mean {
+		intVal /= int64(len(points))
+	}
+	return &metricspb.Point_Int64Value{Int64Value: intVal}
+}
+
+// computeDouble attempts to merge double points into one point based on the provided aggrType
+// Returns the aggregated double value
+func (mtp *metricsTransformProcessor) computeDouble(points []*metricspb.Point, aggrType AggregationType) *metricspb.Point_DoubleValue {
+	doubleVal := float64(0)
+	if len(points) > 0 {
+		doubleVal = points[0].GetDoubleValue()
+	}
+	for _, p := range points[1:] {
+		if aggrType == Sum || aggrType == Mean {
+			doubleVal += p.GetDoubleValue()
+		} else if aggrType == Max {
+			doubleVal = math.Max(doubleVal, p.GetDoubleValue())
+		} else if aggrType == Min {
+			doubleVal = math.Min(doubleVal, p.GetDoubleValue())
+		}
+	}
+	if aggrType == Mean {
+		doubleVal /= float64(len(points))
+	}
+	return &metricspb.Point_DoubleValue{DoubleValue: doubleVal}
+}
+
+// computeDistribution attempts to merge distribution points into one point
+// Returns the aggregated distribution value or values if there are mismatching bounds
+func (mtp *metricsTransformProcessor) computeDistribution(points []*metricspb.Point) []*metricspb.Point_DistributionValue {
+	distVals := make([]*metricspb.Point_DistributionValue, 0)
+	boundsToPoints := mtp.groupPointsByBounds(points)
+	for _, v := range boundsToPoints {
+		var distVal *metricspb.DistributionValue
+		for _, p := range v {
+			if distVal == nil {
+				distVal = p.GetDistributionValue()
+				continue
+			}
+			distVal = mtp.computeDistVals(distVal, p.GetDistributionValue())
+		}
+		distVals = append(distVals, &metricspb.Point_DistributionValue{DistributionValue: distVal})
+	}
+	return distVals
+}
+
+// groupPointsByBounds groups distribution value points by the bounds to further determine the aggregatability because distribution value points can only be aggregated if they have match bounds
+// Returns a map with groups of aggregatable distribution value points
+func (mtp *metricsTransformProcessor) groupPointsByBounds(points []*metricspb.Point) map[string][]*metricspb.Point {
+	boundsToPoints := make(map[string][]*metricspb.Point)
+	for _, p := range points {
+		boundsKey := mtp.boundsToString(p.GetDistributionValue().BucketOptions.GetExplicit().Bounds)
+		if groupedPoints, ok := boundsToPoints[boundsKey]; ok {
+			boundsToPoints[boundsKey] = append(groupedPoints, p)
+		} else {
+			boundsToPoints[boundsKey] = []*metricspb.Point{p}
+		}
+	}
+	if len(boundsToPoints) > 1 {
```

Comment:
1. Is it ever expected that different points could have different distributions? I wouldn't have thought so. It actually feels like a bit of a flaw in the proto design that this is even possible (duplicated data). Might be worth creating an issue in the [proto repo](https://github.com/open-telemetry/opentelemetry-proto/issues) to ask why this is the case (i.e. this could be defined against the Metric itself instead).
2. If this function only ever returns one value, this feels more like a validation, so it could raise an error instead of returning a map. Would it be reasonable for us to not do this validation at all?

Reply:
So based on my initial conversation with Dave, he mentioned to only aggregate distribution points if they have the same bounds, so this is basically my way to account for this condition. However, I am not quite familiar with how it should be conceptually for timeseries and data points. Do you think I can just safely assume that if distribution points have matching labels and even timestamps, they will have the same bound? This sounds very reasonable to me but I don't want to make that assumption by my own reasoning.